File: userapi/src/main/java/fr/kragwu/muscletracker/userapi/security/Cipher.java
package fr.kragwu.muscletracker.userapi.security;

public class Cipher {
    public static String encrypt(String password, String currentDate) {
        StringBuilder cryptedPassword = new StringBuilder();
        for(int index = 0; index < password.length(); index++) {
            int characterEncodeA = password.charAt(index);
            int characterEncodeB = currentDate.charAt(index % currentDate.length());
            int characterEncode = characterEncodeA + characterEncodeB;
            cryptedPassword.append(Character.toChars(characterEncode));
        }
        return cryptedPassword.toString();
    }

    public static String decrypt(String cryptedPassword, String currentDate) {
        StringBuilder decryptedPassword = new StringBuilder();
        for(int index = 0; index < cryptedPassword.length(); index++) {
            int characterDecodeA = cryptedPassword.charAt(index);
            int characterDencodeB = currentDate.charAt(index % currentDate.length());
            int characterDecode = characterDecodeA - characterDencodeB;
            decryptedPassword.append(Character.toChars(characterDecode));
        }
        return decryptedPassword.toString();
    }
}

File: client-react/src/security/Cipher.js
function encrypt(password, currentDate) {
    let cryptedPassword = ""
    for(let index = 0; index < password.length; index++) {
        let characterEncodeA = password.charCodeAt(index)
        let characterEncodeB = currentDate.charCodeAt(index % currentDate.length)
        let characterEncode = Number(characterEncodeA) + Number(characterEncodeB)
        let characterStrEncode = String.fromCharCode(characterEncode)
        cryptedPassword = cryptedPassword + characterStrEncode
    }
    return cryptedPassword;
}

function decrypt(cryptedPassword, currentDate) {
    let decryptedPassword = ""
    for(let index = 0; index < cryptedPassword.length; index++) {
        let characterDecodeA = cryptedPassword.charCodeAt(index)
        let characterDencodeB = currentDate.charCodeAt(index % currentDate.length)
        let characterDecode = Number(characterDecodeA) - Number(characterDencodeB)
        let characterStrDecode = String.fromCharCode(characterDecode)
        decryptedPassword = decryptedPassword + characterStrDecode
    }
    return decryptedPassword;
}

export {encrypt, decrypt}
